lib/src/render.rs

```diff
@@ -157,3 +157,64 @@ where
 {
     RcDoc::intersperse(docs, RcDoc::concat(separator))
 }
+
+/// puts [doc] in a section or a module (that depends on [kind])
+pub(crate) fn enclose<'a, K, U>(kind: K, name: U, ty_context: &Vec<U>, doc: Doc<'a>) -> Doc<'a>
+where
+    U: Into<std::borrow::Cow<'a, str>> + std::marker::Copy,
+    K: Into<std::borrow::Cow<'a, str>>,
+{
+    group([
+        group([text(kind), line(), text(name), text(".")]),
+        nest([
+            hardline(),
+            if ty_context.is_empty() {
+                nil()
+            } else {
+                let types: Vec<Doc<'a>> = ty_context.iter().map(|&ty| text(ty)).collect();
+                group([
+                    nest([
+                        text("Context"),
+                        line(),
+                        text("["),
+                        intersperse(types, [line()]),
+                        line(),
+                        text(": Set]."),
```

Comment:
```suggestion
                        text("{"),
                        intersperse(types, [line()]),
                        line(),
                        text(": Set}."),
```
using `{ ... }` for implicit for uniformity with the rest, if it works.

Reply:
I checked that neither `{ ... }` nor `( ... )` work for `generics/generics.v`. At least not instantly.

Reply:
So I did that because I thought it will be faster that way.

Reply:
but now I see `{ ... }` work when I prepend the items with `@` in the definitions of aliases,
so I'll modify it this way